### ISSUE TITLE
Update ROOT Writers with new discriminated point properties

### ISSUE THREAD

**Issue by cclaessens** [project8/katydid#136]

**Comment by guiguem** [comment]
- [ ] ROOTTreeWriter -> TDiscriminatedPoints1DData
   - fMean
   - fVariance
   - fNeighborhoodAmplitude
- [ ] ROOTTreeWriter -> TSparseWaterfallCandidateData
   - fMinimumTimeInRunC
   - fMaximumTimeInRunC
   - fMinimumTimeInAcq
   - fMaximumTimeInAcq
   - fSlope

**Comment by guiguem** [comment]
It seems that to be able to save Mean, NeighboorhoddAmplitude and Variance in TSparseWaterfallCandidateData, I will have to explode the TGraph2D into a tree.

**Comment by guiguem** [comment]
Looking into the MultiTrackEventData example, I will have to:
- [ ] Create a TClonesArray in the TSparseWaterfallCandidateData structure
- [ ] Add a new structure (TDiscriminatedPoint) for the KTDiscriminatedPoint into KTROOTTreeTypeWriterEventAnalysis.hh which would contain the discriminated points characteristics
- [ ] Replace TGraph2D with a TTree in the Setup
    - [ ] Replace ` fSparseWaterfallCandidateData.fPoints = new TGraph2D(points.size());` (line 334)
by something like `        fSparseWaterfallCandidateData.fPoints = new TTree(...);```
    - [ ] Set the branch addresses for this sub-tree
    - [ ] Fill the tree using fSparseWaterfallCandidateData.fPoints
- [ ] Done?

**Comment by nsoblath** [comment]
I don't think you replace the TGraph2D with a TTree, but with a branch.  That effectively gives you a sub-tree.  I can look into this with you later.

**Comment by cclaessens** [comment]
I think we also want to add writing of sequential lines. So that optimization can look at all the intermediate processing steps.

**Comment by guiguem** [comment]
@cclaessens Can you write down the variables of the SequentialLineData you want to be saved in the ROOT file?
It seems to me that SumX and so on are not interesting to have, and I would only save variable with names similar to the SparseWaterfallCandidateData so:
```
            MEMBERVARIABLE(unsigned, Component);
            MEMBERVARIABLE(uint64_t, AcquisitionID);
            MEMBERVARIABLE(unsigned, CandidateID);

            MEMBERVARIABLE(double, TimeInRunC);
            MEMBERVARIABLE(double, TimeLength);
            MEMBERVARIABLE(double, TimeInAcq);
            MEMBERVARIABLE(double, MinFrequency);
            MEMBERVARIABLE(double, MaxFrequency);
            MEMBERVARIABLE(double, FrequencyWidth);
```
Note that you are not creating the same variables there to, is there a need for having different variables output (like StartFrequency instead of MinFrequency)?
Maybe we should unify what we can?

**Comment by guiguem** [comment]
Solved in 45d72130a3e1aa9ade3ef4f73b8263cc135f2bb2
